### PR TITLE
Update docs build requirements to most recent versions of dependencies

### PR DESCRIPTION
**Pull request by samccann**
##### SUMMARY

This updates to Sphinx 4.2, antsibull 0.39.2 and docutils 0.17.1 (highest that sphinx supports).
It also aligns docs CI tests with docs builds as they'd gottten out of sync

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

docs/docsite/known_good_reqs.txt
docs/docsite/requirements.txt
test/sanity/code-smell/docs-build.requirements.txt
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

**Comment by jborean93** [comment]
Once CI has passed and the PR is merged these requirements will have to be updated in https://github.com/ansible/ansible-core-test-container so our test container doesn't need to update the requirements on every run. Are you also able to add a changelog fragment to the PR?

File: test/sanity/code-smell/docs-build.requirements.txt [hunk 1]
```diff
@@ -1,11 +1,11 @@
 jinja2 == 3.0.1
 pyyaml == 5.4.1
 resolvelib == 0.5.4
-sphinx == 2.1.2
-sphinx-notfound-page == 0.7.1
+sphinx == 4.2.0
+sphinx-notfound-page == 0.8
```
Comment (mattclay) [inline test/sanity/code-smell/docs-build.requirements.txt]: The sphinx requirements for `rstcheck` should be updated to match as well:

https://github.com/ansible/ansible/blob/devel/test/sanity/code-smell/rstcheck.requirements.txt

**Comment by samccann** [comment]
@jborean93 Thanks for the link. I'm not seeing these requirements in the test container repo but maybe I'm looking at the wrong file(s)? I'm looking under /requirements and /freeze in that repo.

**Comment by mattclay** [comment]
@samccann The process is a bit different than it used to be. The requirements don't show up in the repo, but instead the SHA of ansible/ansible is updated with `make update` in the default container. @jborean93 or I can help you with that once this is merged.

**Comment by mattclay** [comment]
@samccann I've updated the sanity test requirements as we discussed, so this PR is ready to be rebased.

**Review by mattclay — APPROVED** [review]

**Pull request merged**